File: app/src/main/java/com/yc/emotion/home/utils/DateUtils.java
package com.yc.emotion.home.utils;

import android.text.TextUtils;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

/**
 * Created by admin on 2018/2/12.
 */

public class DateUtils {
    public static String getFormatDateInSecond(String second) {
        if (!TextUtils.isEmpty(second)) {
            int seconds = Integer.parseInt(second);
            int temp = 0;
            StringBuffer sb = new StringBuffer();
            temp = seconds / 60;
            sb.append((temp < 10) ? "0" + temp + ":" : "" + temp + ":");

            temp = seconds % 60;
            sb.append((temp < 10) ? "0" + temp : "" + temp);

            return sb.toString();
        }
        return null;
    }

    public static String formatTimeToStr(long time) {
        return formatTimeToStr(time ,"yyyy-MM-dd HH:mm");
    }

    public static String formatTimeToStr(long time,String format) {
        SimpleDateFormat simpleDateFormat = new SimpleDateFormat(format, Locale.getDefault());
        return simpleDateFormat.format(new Date(time * 1000));
    }

}

File: app/src/main/java/com/yc/emotion/home/utils/DateUtils.kt
package com.yc.emotion.home.utils

import android.text.TextUtils
import java.text.SimpleDateFormat
import java.util.*

/**
 * Created by admin on 2018/2/12.
 */
object DateUtils {
    fun getFormatDateInSecond(second: String): String? {
        if (!TextUtils.isEmpty(second)) {
            val seconds = second.toInt()
            var temp = 0
            val sb = StringBuffer()
            temp = seconds / 60
            sb.append(if (temp < 10) "0$temp:" else "$temp:")
            temp = seconds % 60
            sb.append(if (temp < 10) "0$temp" else "" + temp)
            return sb.toString()
        }
        return null
    }

    @JvmOverloads
    fun formatTimeToStr(time: Long, format: String? = "yyyy-MM-dd HH:mm"): String {
        val simpleDateFormat = SimpleDateFormat(format, Locale.getDefault())
        return simpleDateFormat.format(Date(time * 1000))
    }
}